### ISSUE TITLE
Delay-send / "Send later" / Scheduled sending

### ISSUE THREAD

**Issue by nekohayo** [cypht-org/cypht#576]
Issue #572 talks about snoozing (i.e. hiding a received email from the inbox for a certain amount of time), but what I'd like to suggest here is a different feature. The usecase for me is this:

*  Business or events-related outreach emails typically have a context/timing-sensitive aspect, so I'd like to be able to schedule them (especially considering the timezone of the person I'm contacting)

* When dealing with public-facing businesses, it is often counterproductive to send emails on certain days or hours (ex: Friday afternoons, when you know everything is likely to be forgotten over the week-end), so you want to schedule your email for a better time during business hours (ex: Tuesday morning).

* Human-centric staff managers and directors like me know that it is better to schedule mails to staffers to their business hours, avoiding emails that would arrive outside work hours and cause undue stress or cognitive load. Basically, mindfulness for staffers' "[right to disconnect](https://en.wikipedia.org/wiki/Right_to_disconnect)".

The corresponding issue in my local mail client (where some technical approaches are mentioned) is https://gitlab.gnome.org/GNOME/evolution/-/issues/411#note_874473

**Comment by marclaporte** [comment]
See also: https://github.com/jasonmunro/cypht/issues/641

**Comment by dumblob** [comment]
Yeah, I outlined an envisioned potential implementation in #641 . Feel free to take a look.

**Comment by dumblob** [comment]
Regarding the proposal of the "local approach" in https://gitlab.gnome.org/GNOME/evolution/-/issues/411#note_874473 I do not think it is viable for Cypht. It simply has to be "fire and forget" - i.e. closing the web site (incl. suspending the web browser tab) must not hinder this functionality in any way.

**Comment by nekohayo** [comment]
A bit like the "snooze" counterpart, I've requested clarification on JMAP side at https://github.com/jmapio/jmap/issues/373 because it is unclear to me (perhaps because I'm not familiar with such things as protocols and specs and whatnot)

**Comment by marclaporte** [comment]
Is this support by IMAP/SMTP?

If not (how) can this be done without a cron job?

**Comment by marclaporte** [comment]
Some ideas from Roundcube: https://github.com/roundcube/roundcubemail/issues/8110

**Comment by marclaporte** [comment]
Much to my surprise, this issue is number 3 for most reactions:
https://github.com/cypht-org/cypht/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions

**Comment by marclaporte** [comment]
Ok, let's do this with the proposed standards for JMAP and SMTP. Let's not try to do this in a non-standard way as it would be a lot of work:
# We would need a cron job
# Cypht doesn't know the SMTP password if user is not logged in: https://www.cypht.org/security.html

Step 1: Find mail servers that implement https://www.rfc-editor.org/rfc/rfc4865 and sendAt https://www.rfc-editor.org/rfc/rfc8621.html#page-76
Step 2: Code it :-)

**Comment by marclaporte** [comment]
Done!
https://github.com/cypht-org/cypht/pull/1091
